test_runner/performance/test_perf_ingest_using_pgcopydb.py

```diff
@@ -0,0 +1,343 @@
+    - PG_CONFIG: e.g. /tmp/neon/pg_install/v16/bin/pg_config
+    - PSQL: e.g. /tmp/neon/pg_install/v16/bin/psql
+    - PG_16_LIB_PATH: e.g. /tmp/neon/pg_install/v16/lib
+    - PGCOPYDB: e.g. /pgcopydb/bin/pgcopydb
+    - PGCOPYDB_LIB_PATH: e.g. /pgcopydb/lib
+    - BENCHMARK_INGEST_SOURCE_CONNSTR
+    - BENCHMARK_INGEST_TARGET_CONNSTR
+    - PERF_TEST_RESULT_CONNSTR
+    - TARGET_PROJECT_TYPE
+    - COMMIT_HASH
+
+    """
+    # Ensure required environment variables are set
+    required_env_vars = [
+        "PGCOPYDB",
+        "PGCOPYDB_LIB_PATH",
+        "PG_CONFIG",
+        "PSQL",
+        "PG_16_LIB_PATH",
+        "BENCHMARK_INGEST_SOURCE_CONNSTR",
+        "BENCHMARK_INGEST_TARGET_CONNSTR",
+        "PERF_TEST_RESULT_CONNSTR",
+        "TARGET_PROJECT_TYPE",
+        "COMMIT_HASH",
+    ]
+    for var in required_env_vars:
+        if not os.getenv(var):
+            raise OSError(f"Required environment variable '{var}' is not set.")
+
+    # Set additional environment variables for the pgcopydb command
+    os.environ["LD_LIBRARY_PATH"] = (
+        f"{os.getenv('PGCOPYDB_LIB_PATH')}:{os.getenv('PG_16_LIB_PATH')}"
+    )
+    os.environ["PGCOPYDB_SOURCE_PGURI"] = cast(str, os.getenv("BENCHMARK_INGEST_SOURCE_CONNSTR"))
+    os.environ["PGCOPYDB_TARGET_PGURI"] = cast(str, os.getenv("BENCHMARK_INGEST_TARGET_CONNSTR"))
+    os.environ["PGOPTIONS"] = (
+        "-c maintenance_work_mem=8388608 -c max_parallel_maintenance_workers=7"
+    )
+
+
+def build_pgcopydb_command():
+    """Builds the pgcopydb command to execute using existing environment variables."""
+    pgcopydb_executable = os.getenv("PGCOPYDB")
+    if not pgcopydb_executable:
+        raise OSError("PGCOPYDB environment variable is not set.")
+
+    return [
+        pgcopydb_executable,
+        "clone",
+        "--skip-vacuum",
+        "--no-owner",
+        "--no-acl",
+        "--skip-db-properties",
+        "--table-jobs",
+        "4",
+        "--index-jobs",
+        "4",
+        "--restore-jobs",
+        "4",
+        "--split-tables-larger-than",
+        "10GB",
+        "--skip-extensions",
+        "--use-copy-binary",
+        "--filters",
+        "/tmp/pgcopydb_filter.txt",
+    ]
+
+
+def create_pgcopydb_filter_file():
+    """Creates the /tmp/pgcopydb_filter.txt file required by pgcopydb."""
+    filter_content = """\
+[include-only-table]
+public.events
+public.emails
+public.email_transmissions
+public.payments
+public.editions
+public.edition_modules
+public.sp_content
+public.email_broadcasts
+public.user_collections
+public.devices
+public.user_accounts
+public.lessons
+public.lesson_users
+public.payment_methods
+public.orders
+public.course_emails
+public.modules
+public.users
+public.module_users
+public.courses
+public.payment_gateway_keys
+public.accounts
+public.roles
+public.payment_gateways
+public.management
+public.event_names
+"""
+    filter_path = Path("/tmp/pgcopydb_filter.txt")
```

Comment:
Usually, we create files in [`test_output_dir`](https://github.com/neondatabase/neon/blob/b4e00b8b220b6443c117fbaf9746ab4ef9c38e55/test_runner/fixtures/paths.py#L165-L193), I'd recommend creating any test related file in that directory (you can either convert `create_pgcopydb_filter_file` to a fixture that dependends on `test_output_dir` or pass path as a parameter)